core/modules/config.go

```diff
@@ -13,6 +13,9 @@ const EngineVersionLatest string = "latest"
 
 // ModuleConfig is the config for a single module as loaded from a dagger.json file.
 type ModuleConfig struct {
+	// The self-describing json $schema of
+	Schema string `json:"$schema"`
```

Comment:
Hm, I was hoping to avoid adding this as a property on here. The code that interacts with `ModuleConfig` shouldn't be able to access this. Ideally it could be a private field - but then it can't be accessed by the `json` parser either.

Doing anything else feels bizarrely hacky though, so happy to leave as is. Maybe make a note that this field is *only* for user usage, and shouldn't be modified/accessed?

Reply:
Some alternative would be to do most of the parsing without this schema field.

Then right before writing it back, we can parse out the `$schema` and attach it to a combined struct.

But personally fine with tacking on a comment to not pay any mind to `Schema`

Reply:
Yeah, that feels like a lot of effort :scream: Happy with the comment!